cog_fixtures/cog.py

```diff
@@ -0,0 +1,118 @@
+"""cog_fixtures.cog module"""
+import abc
+from dataclasses import dataclass
+from pathlib import Path
+from typing import Dict, Optional, Sequence, Union
+
+import affine
+import rasterio
+from fake_geo_images.fakegeoimages import FakeGeoImage
+from rasterio.crs import CRS
+from rasterio.io import MemoryFile
+from rio_cogeo.cogeo import cog_translate
+
+
+@dataclass  # type: ignore
+class ImageBase(abc.ABC):
+    """base image class"""
+
+    def __enter__(self):
+        """support context management"""
+        return self
+
+    def __exit__(self, exc_type, exc_val, exc_tb):
+        """support context management"""
+        self.close()
+
+    @abc.abstractmethod
+    def __post_init__(self):
+        """post init hook"""
+        ...
+
+    @abc.abstractmethod
+    def close(self):
+        """close resources"""
+        ...
+
+
+@dataclass
+class FakeImage(ImageBase):
+    """fake rasterio image"""
+
+    driver: str
+    width: int
+    height: int
+    count: int
+    dtype: str
+    crs: CRS
+    transform: affine.Affine
+    nodata: Union[float, int] = -1.0
```

Comment:
why is the nodata set to `-1.0` ?

Reply:
Thats how fake-geo-images specifies to not have a `nodata` value.  I don't think its very intuitive either, maybe we should default to `None`.

Reply:
so how do you specify nodata to be `-1` ?